wfl/Database/MongoDBManager.cs

```diff
@@ -19,62 +20,66 @@ public MongoDBManager()
             db = client.GetDatabase("WFL");
         }
 
-        public async Task<Restaurant> InsertRestaurantAsync(RestaurantDto r)
+        private string GenerateCollectionName<T>() where T: new()
         {
-            IMongoCollection<Restaurant> collection = db.GetCollection<Restaurant>("restaurants");
-            Restaurant restaurant = new Restaurant()
+            T t = new T();
+            string type = t.GetType().ToString();
+            string collectionName = "";
+            bool started = true;
+            foreach(char c in type)
             {
-                Name = r.Name,
-                MenuLink = r.MenuLink
-            };
-            await collection.InsertOneAsync(restaurant);
-            return restaurant;
-        }
+                if(c == '.')
+                {
+                    collectionName = "";
+                    started = false;
+                }
+                else if (char.IsUpper(c))
+                {
+                    if (started == true)
+                    {
+                        collectionName += '-';
+                    }
+                    else
+                        started = true;
+                    collectionName += char.ToLower(c);
+                }
+                else
+                    collectionName += c;
+            }
+            collectionName += 's';
```

Comment:
I agree this would have to change if we have an entity ending in an `s`. I didn't think it was necessary to account for that at this stage because there are currently only three different entities to be in our DB and none of them end in `s`.